engine/engine.go

```diff
@@ -258,11 +278,13 @@ func (e *Engine) Shutdown() error {
 	err := e.publisher.Close()
 	if err != nil {
 		err = fmt.Errorf("error closing leg publisher: %w", err)
+		return err
```

Comment:
This means `e.cache` will not close if we fail to close the publisher. 

I am curious why this change?

Reply:
ah yes, I mean to undo this. Thanks for the catch!